include/spdlog/logger.h

```diff
@@ -44,6 +44,17 @@ class logger
     template <typename Arg1, typename... Args> void error(const char* fmt, const Arg1&, const Args&... args);
     template <typename Arg1, typename... Args> void critical(const char* fmt, const Arg1&, const Args&... args);
 
+#ifdef SPDLOG_ENABLE_LOGMSG_METADATA
+    template <typename... Args> void log(level::level_enum lvl, attrmap_type& ma, const char* fmt, const Args&... args);
+    template <typename... Args> void log(level::level_enum lvl, attrmap_type& ma, const char* msg);
```

Comment:
shouldn't these all be `attrmap_type&&`? Since you're moving from `ma` inside the implementations.